user/views.py

```diff
@@ -12,3 +16,41 @@ class SignUpView(CreateView):
     # プロジェクトのURLConf(url.py)が読み込まれる前に、URLを返す
     success_url = reverse_lazy('twitter:home')
     template_name = 'user/signup.html'
+
+    def form_valid(self, form):  # 追記
+        response = super().form_valid(form)
+        user = self.object
+        login(self.request, user)
+        return response
+
+
+class Login(LoginView):
+    form_class = LoginForm
+    template_name = 'user/login.html'
+
+
+class LogoutConfirmView(LoginRequiredMixin, TemplateView):
+    template_name = 'user/logout_confirm.html'
+
+
+class ProfileDisplay(LoginRequiredMixin, TemplateView):
+    template_name = 'user/profile.html'
+
+
+class OnlyYouMixin(UserPassesTestMixin):
+    raise_exception = False     # set True if raise 403_Forbidden
+
+    def test_func(self):
+        user = self.request.user
+        return user.pk == self.kwargs['pk'] or user.is_superuser
+
+
+class ProfileUpdateView(LoginRequiredMixin, OnlyYouMixin, UpdateView):
+    model = get_user_model()
+    template_name = 'user/profile_update.html'
+    fields = ('nickname', 'introduction')
+    success_url = reverse_lazy('user:profile')
+
+    def get_object(self):
+        self.kwargs['pk'] = self.request.user.pk
+        return super().get_object()
```

Comment:
get_objectは，関数名の通り，返り値でDBから特定のレコードのobjectを指定してあげられる関数です。

この部分で，セッションからuserのpkを直接取得しているので，`urls.py`で
https://github.com/amoon46/TwitterClone/blob/69af81267ce014c7b903c31896c204c1f6e61212/user/urls.py#L14
のように書いていますが，urlにpkを書いている意味がないですね。

ここで，
```suggestion
    def get_object(self):
        return self.request.user
```
としてuserのobjectをそのまま返してあげれば，urls.pyのpkが完全に不要になります。
追加で，`update/`だと何をupdateするのかよくわからないので，いい感じのurlに変更してください！（前回抜けててすみません）

付随して，「url変えれば別のユーザーとして更新できてしまわないように」という意図のtestですが，原理的に「存在しないユーザーと他のユーザーのテスト」もこの実装なら不要になるので書かなくてOKです！

Reply:
urlsの<int:pk>を削除してしまうとviewにおけるOnlyYouMixinのtest_funcがうまく機能しなくなるのですが、そちらはどのように対応したらいいでしょうか？

Reply:
OnlyYouMixinは削除しても，Login必須ということはログイン中のユーザー本人（self.request.user）が必ず返されるので実質的に同じ意味になるので無くても良いんじゃないかな？と思います

---
url内にユーザーを指定する何かがあった方が良いともし思った場合は，
`profile/`を`user/profile/`とすると，ログイン中のユーザーであることを明記できます。（`user/`と単数で指定した場合はログイン中のユーザーを指すことが多いようです）
もし別のユーザーのprofileを見たい場合は，
`users/<pk>/profile/`のように指定するのが一般的です。（たくさんあるusersの中から，このユーザー！と1つ絞り込むイメージ）